### DISCUSSION TITLE
Avalonia obfuscator

### DISCUSSION THREAD

**Discussion by vadimaus** [AvaloniaUI/Avalonia#10887] [Q&A]
Hello

Is there a good **Obfuscator** for Avalonia?

Thanks!

**Comment by timunie** [comment]
I think this is not a task for Avalonia, as Avalonia is basically an UI library for dotNET. If there is any for plain dotNET, I think it should also work for Avalonia.

**Reply by vadimaus** [reply]
Hello @timunie 
Not saying that this is an Avalonia task, but wondering if there are some obfuscators that are designed specifically for Avalonia.
I did try couple of .NET ones and they didn't work. They did obfuscate but the app would crash on startup.
That's why asked, in hope that someone will point to the right resource..

**Reply by maxkatz6** [reply]
> if there are some obfuscators that are designed specifically for Avalonia

There is no need for that. Avalonia XAML is translated into IL code, so obfuscation will work.

The best "obfuscation" would be [Native AOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/).
Other than that, any decent obfuscation should work.

Note, in 0.10 version you might need to configure some rules (and I believe that's the problem you saw). For Obfuscar I used this rule set: https://gist.github.com/maxkatz6/b35e1a44bc1d607cbfc3f1b218f58904
In 11.0 previews it's not necessary.

**Reply by vadimaus** [reply]
Thank you a lot! Much appreciated.
